src/sardana/taurus/core/tango/sardana/macro.py

```diff
@@ -1301,45 +1301,60 @@ def createMacroNode(macro_name, params_def, macro_params):
-        # this ignores raw_parameters which exceeds the param_def
+        param_idx = 0
         for param_node, param_raw in zip(param_nodes, macro_params):
             if isinstance(param_node, SingleParamNode):
                 param_node.setValue(param_raw)
-            # the rest of the values are interpreted as repeat parameter
-            elif isinstance(param_node, RepeatParamNode):
+            # Repeat parameters that are not at the end.
+            elif (isinstance(param_node, RepeatParamNode) and
+                  param_idx < (len_param_nodes - 1)):
+                repeat_node = param_node.child(0)
+                # Add a new repeat node. This is needed when the raw values
+                # fill more repeat nodes that the minimum number of
+                # repetitions e.g. min=0.
+                if repeat_node is None:
+                    repeat_node = param_node.addRepeat()
+                if len(repeat_node.children()) > 1:
+                    msg = ("repeat parameter with more than one member must "
+                           "be defined at the end when intended to "
+                           "be used with spock syntax")
+                    raise Exception(msg)
+                member_node = repeat_node.child(0)
+                if isinstance(member_node, RepeatParamNode):
+                    msg = ("nested repeat parameters are not allowed when "
+                           "intended to be used with spock syntax")
+                    raise Exception(msg)
+                member_node.setValue(param_raw)
+            # The resting values are interpreted as repeat parameter values.
+            # This ignores parameter values which exceeds the parameters
+            # definition.
+            else:
                 params_info = param_node.paramsInfo()
                 params_info_len = len(params_info)
+                last_param_idx = len_param_nodes - 1
                 rep = 0
                 mem = 0
-                rest_raw = macro_params[i:]
+                rest_raw = macro_params[last_param_idx:]
                 for member_raw in rest_raw:
                     repeat_node = param_node.child(rep)
-                    # add a new repeat node (this is needed when the raw values
-                    # fill more repeat nodes that the minimum number of
-                    # repetitions e.g. min=0
+                    # Add a new repeat node. This is needed when the raw
+                    # values fill more repeat nodes that the minimum number
+                    # of repetitions e.g. min=0.
                     if repeat_node is None:
                         repeat_node = param_node.addRepeat()
                     member_node = repeat_node.child(mem)
                     if isinstance(member_node, RepeatParamNode):
-                        msg = ("Nested repeat parameters are not allowed")
+                        msg = ("nested repeat parameters are not allowed "
```

Comment:
I would suggest the following message instead:
`"Nested parameters not allowed in (old) spock syntax. Use "bracket" syntax instead."`